src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionPal.Windows.cs

```diff
@@ -164,6 +164,15 @@ public static unsafe Task GetAddrInfoAsync(string hostName, bool justAddresses,
             {
                 state.RegisterForCancellation(cancellationToken);
             }
+            else if (errorCode == SocketError.TryAgain)
+            {
+                // WSATRY_AGAIN indicates possible problem with reachability according to docs.
+                // However, if servers are really unreachable, we would still get IOPenfing here
+                // and final result would be posted via overlapped IO.
+                // synchronous failure here may signal issue when GetAddrInfoExW does not work from
+                // impersonificated context.
+                return Task.FromException(new InvalidOperationException());
+            }
```

Comment:
Also, using a dummy exception here is strange.  I think this would be better as a Try method, e.g.
```C#
public static unsafe bool TryGetAddrInfoAsync(string hostName, bool justAddresses, AddressFamily family, CancellationToken cancellationToken, out Task task)
```
or something along those lines.

Reply:
updated. I was thinking about changing return type but decided not to do it at the end.